src/app/modals/LoginModal/LoginModal.tsx

```diff
@@ -13,46 +15,73 @@ interface LoginModalProps {
 
 const LoginModal: React.FC<LoginModalProps> = ({ isOpen, onClose }) => {
   const { login } = useAuth();
+  const [error, setError] = useState<string | null>(null);
 
-  const [username, setUsername] = useState("");
-  const [password, setPassword] = useState("");
+  const initialValues = {
+    username: "",
+    password: "",
+  };
+
+  const validationSchema = Yup.object().shape({
+    username: Yup.string().required("Username is required"),
+    password: Yup.string().required("Password is required"),
+  });
 
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    login(username, password);
-    onClose();
+  const handleSubmit = async (values: { username: string; password: string }) => {
```

Comment:
Formik'in `setError` gibi bir parametresi olmasi lazim, onu kullanarak error handling yapabilirsin. State olarak tuttugun error'u kaldiralim.